frontend/pkg/frontend/frontend.go

```diff
@@ -511,11 +504,21 @@ func (f *Frontend) ArmSubscriptionPut(writer http.ResponseWriter, request *http.
 		doc.Subscription = &subscription
 	}
 
+	// Log differences
+	f.logSubscriptionDifferences(doc.Subscription, &subscription)
+
 	err = f.dbClient.SetSubscriptionDoc(ctx, doc)
 	if err != nil {
 		f.logger.Error("failed to create document for subscription %s: %v", subscriptionID, err)
 	}
 
+	// Emit the subscription state metric
+	f.metrics.EmitGauge("subscription_lifecycle", 1, map[string]string{
```

Comment:
I know it was done like this before, but at some point it could be a good idea to extract all the "subscription_lifecycle" and other metric names to constants on top of the file (or in a different file).

Reply:
Another possibility is to wrapper `EmitGauge` with more specific use cases.  Like this case might be:
```
func (pe *PrometheusEmitter) EmitSubscriptionLifecycle(region, subscriptionID string, state arm.RegistrationState) {
        pe.EmitGauge("subscription_lifecycle", 1, map[string]string{
                "region":         region,
                "subscriptionid": subscriptionID,
                "state":          string(state),
        })
}
```